lib/scripts/diffExperimentBuilds.py

```diff
@@ -0,0 +1,226 @@
+
+sys.path.append(str(benchpark.paths.benchpark_home) + "/spack/lib/spack")
+import llnl.util.tty.color as color  # noqa: E402
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="""Compare YAMLs generated from 'system init' between two different commits of Benchpark.
+        (check if the output of system init was changed between commits).
+
+        Examples:
+            - benchpark-python diffExperimentBuilds.py -s llnl-cluster -c ruby -p openmp
+            - benchpark-python diffExperimentBuilds.py -s llnl-sierra -p cuda
+            - benchpark-python diffExperimentBuilds.py -s llnl-elcapitan -c tioga -p rocm""",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument(
+        "-s",
+        "--system",
+        type=str,
+        required=True,
+        help="Name of system.py to use for the experiment. This should be the system the script is being ran on. See 'benchpark list systems'. (e.g. 'llnl-cluster')",
+    )
+    parser.add_argument(
+        "-p",
+        "--programming-model",
+        type=str,
+        required=True,
+        help="Programming model to run experiments. choose from: ('openmp', 'cuda', 'rocm')",
+    )
+    parser.add_argument(
+        "--commit-hash1",
+        type=str,
+        default="develop",
+        help="Commit hash or branch name for the version of Benchpark to compare against (default: develop).",
+    )
+    parser.add_argument(
+        "--commit-hash2",
+        type=str,
+        default="pearce8-patch-3",
+        help="Second commit hash or branch name for the newer version of Benchpark",
+    )
+    parser.add_argument(
+        "-c",
+        "--cluster",
+        type=str,
+        default=None,
+        help="System variant if applicable. This should be the cluster the script is being ran on. See 'benchpark list systems'. (e.g. 'ruby' for 'llnl-cluster')",
+    )
+    parser.add_argument(
+        "--extra-spec",
+        default="",
+        type=str,
+        help="provide additional spec to experiment spec string",
+    )
+    parser.add_argument(
+        "-e",
+        "--experiments",
+        nargs="*",
+        default=[],
+    )
+    parser.add_argument("--run-experiment", action="store_true")
+
+    args = parser.parse_args()
+
+    old_name = f"benchpark-{args.commit_hash1}-{args.programming_model}"
+    new_name = f"benchpark-{args.commit_hash2}-{args.programming_model}"
+
+    if args.experiments == []:
+        replace_dict = {
+            " ": "",
+            "Experiments:": "",
+            "/": "+",
+            "OpenMPExperiment": "openmp",
+            "CudaExperiment": "cuda",
+            "ROCmExperiment": "rocm",
+        }
+        experiments_out = subprocess.run(
+            ["benchpark", "list", "experiments"], text=True, capture_output=True
+        )
+        experiments = experiments_out.stdout
+        for k, v in replace_dict.items():
+            experiments = experiments.replace(k, v)
```

Comment:
This section converts `benchpark list experiments` output, i.e.:
```
    qws/OpenMPExperiment
    raja-perf/CudaExperiment
    raja-perf/ROCmExperiment
    raja-perf/OpenMPExperiment
    ...
```
to
```
    qws+openmp
    raja-perf+cuda
    raja-perf+rocm
    raja-perf+openmp
    ...
```

Would we rather the output of `benchpark list experiments` to just be the latter (with the spec)?

Reply:
Yes, `benchpark list experiments` should list::

```
qws+openmp 
raja-perf+cuda 
raja-perf+rocm 
raja-perf+openmp
```

`benchpark list benchmarks` should list::

```
qws
raja-perf
```

Reply:
Done in [c4743b8](https://github.com/LLNL/benchpark/pull/655/commits/c4743b85d155dd2edc3da1038e90e8aa44372112)